src/AbstractSessionGenerator.php

```diff
@@ -87,23 +91,22 @@ protected function _generateRecursive(
                     if ($validator !== null) {
                         $validator->validate($session);
                     }
-                    $results[]          = $session;
-                    $processed[$_start] = true;
+                    $results[] = $session;
```

Comment:
At first, thought this may have an undesirable memory footprint. Then realized that the session object is shared, and is already "cached" in this scope. Good job!